MigrationAssistant - Feature: Set desired count for proxies (#10082)
opensearch-project/documentation-website: _migration-assistant/deploying-migration-assistant/configuration-options.md

            "passwordFromSecretArn": <TARGET_CLUSTER_PASSWORD_SECRET>
        }
    },

>     "// settingsForCaptureAndReplay": "Enable the below services for live traffic capture and replay",

Reviewer: ```suggestion
    "// settingsForCaptureAndReplay": "Enable the following services for live traffic capture and replay",
```

>     "// settingsForCaptureAndReplay": "Enable the following services for live traffic capture and replay",

Author: Once we merge this, i'll merge a correction int he sample CDK to match this exactly

    "trafficReplayerServiceEnabled": true,

>     "// help trafficReplayerExtraArgs": "Increase speedup factor in order replay requests at a faster rate to catchup",

Reviewer 2: ```suggestion
    "// help trafficReplayerExtraArgs": "Increase speedup factor in order replay requests at a faster rate to catch up",
```

    "trafficReplayerExtraArgs": "--speedup-factor 1.5",

>     "// help capture/target proxy pt. 1 of 2": "captureProxyService and targetClusterProxyService require networking access configured to successfully deploy,",

Reviewer 2: ```suggestion
    "// help capture/target proxy pt. 1 of 2": "captureProxyService and targetClusterProxyService require networking access to be configured in order to successfully deploy",
```

>     "// help capture/target proxy pt. 2 of 2": "consider deploying without first and enabling after ensuring cluster networking access on the migration console",

Reviewer 2: Please revise for clarity. Deploying what first? Ensuring who/what has cluster networking access?

Author: updated this

>     "// help capture/target proxy pt. 2 of 2": "consider deploying without first and enabling after ensuring cluster networking access on the migration console",

Reviewer: @AndreKurait: can you verify that this revision retains the correct technical meaning?

```suggestion
    "// help capture/target proxy pt. 2 of 2": "Consider deploying without the proxy first. After verifying cluster network access in the migration console, you can enable the proxy.",
```

| :--- | :--- | :--- |
| `captureProxyServiceEnabled`    | `true`  | Enables the Capture Proxy service deployment using an AWS CloudFormation stack.  |
| `captureProxyExtraArgs`  | `"--suppressCaptureForHeaderMatch user-agent .*elastic-java/7.17.0.*"`  | Extra arguments for the Capture Proxy command, including options specified by the [Capture Proxy](https://github.com/opensearch-project/opensearch-migrations/blob/main/TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/CaptureProxy.java).  |

> | `captureProxyDesiredCount`  | `3`  | Set the number of capture proxy Amazon ECS tasks. In most cases, keep this at 0 until after verifying connectivity with the source and target cluster in the migration console. After deploying the networking setup can be modified to allow ingress into the existing cluster security groups by the migration security groups.  |

Reviewer: ```suggestion
| `captureProxyDesiredCount`  | `3`  |  Sets the number of capture proxy Amazon ECS tasks. In most cases, keep this setting at `0` until you verify connectivity between the source and target clusters in the migration console. After deployment, you can modify the networking setup to allow ingress from the migration security groups into the existing cluster security groups.  |
```

> | `captureProxyDesiredCount`  | `3`  |  Sets the number of capture proxy Amazon ECS tasks. In most cases, keep this setting at `0` until you verify connectivity between the source and target clusters in the migration console. After deployment, you can modify the networking setup to allow ingress from the migration security groups into the existing cluster security groups.  |

Reviewer 2: ```suggestion
| `captureProxyDesiredCount`  | `3`  |  Sets the number of capture proxy Amazon Elastic Container Service (Amazon ECS) tasks. In most cases, keep this setting at `0` until you verify connectivity between the source and target clusters in the migration console. After deployment, you can modify the networking setup to allow ingress from the migration security groups into the existing cluster security groups.  |
```

| `trafficReplayerServiceEnabled` | `true`  | Enables the Traffic Replayer service deployment using a CloudFormation stack.  |
| `trafficReplayerExtraArgs`      | `"--sigv4-auth-header-service-region es,us-east-1 --speedup-factor 5"`                 | Extra arguments for the Traffic Replayer command, including options for auth headers and other parameters specified by the [Traffic Replayer](https://github.com/opensearch-project/opensearch-migrations/blob/main/TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java). |
| `targetClusterProxyServiceEnabled` | `true` | Enables the target cluster proxy service deployment using a CloudFormation stack. |

> | `targetClusterProxyDesiredCount`  | `3`  | Set the number of target cluster proxy Amazon ECS tasks. In most cases, keep this setting at `0` until after verifying connectivity with the source and target cluster in the migration console. After deploying the networking setup can be modified to allow ingress into the existing cluster security groups by the migration security groups.  |

Reviewer: ```suggestion
| `targetClusterProxyDesiredCount`  | `3`  | Set the number of target cluster proxy Amazon ECS tasks. In most cases, keep this setting at `0` until you verify connectivity between the source and target clusters in the migration console. After deployment, you can modify the networking setup to allow ingress from the migration security groups into the existing cluster security groups.  |
```

> | `targetClusterProxyDesiredCount`  | `3`  | Set the number of target cluster proxy ecs tasks. In many cases it can be a good idea to keep this at 0 until after verifying connectivity with the source and target cluster in the migration console. After deploying the networking setup can be modified to allow ingress into existing cluster security groups by the migration security groups.  |

Reviewer: ```suggestion
| `targetClusterProxyDesiredCount`  | `3`  | Set the number of target cluster proxy Amazon ECS tasks. In most cases, keep this setting at `0` until after verifying connectivity with the source and target cluster in the migration console. After deploying the networking setup can be modified to allow ingress into the existing cluster security groups by the migration security groups.  |
```

> | `targetClusterProxyDesiredCount`  | `3`  | Sets the number of target cluster proxy Amazon ECS tasks. In most cases, keep this setting at `0` until you verify connectivity between the source and target clusters in the migration console. After deployment, you can modify the networking setup to allow ingress from the migration security groups into the existing cluster security groups.  |

Author: We have not performed load tests yet on the proxy throughput, `3` adds 1 per AZ. Customer will need to determine the scale depending on their workload. `3` here is listed as a `Example` and not any recommendation

> | `targetClusterProxyDesiredCount`  | `3`  | Sets the number of target cluster proxy Amazon ECS tasks. In most cases, keep this setting at `0` until you verify connectivity between the source and target clusters in the migration console. After deployment, you can modify the networking setup to allow ingress from the migration security groups into the existing cluster security groups.  |

Reviewer 3: It sounds like there are details missing, it sounds like this value should be 0 or 3 == the number of AZs in the region, and anything else is basically debugging/troubleshooting for us?

What do you think about making an update to this effect?
